app/src/main/java/org/stepic/droid/core/presenters/StepAttemptPresenter.kt

```diff
@@ -139,17 +139,15 @@ class StepAttemptPresenter
             worker?.schedule(
                     Runnable {
                         try {
-                            var submissionsResponse = api.getSubmissionForStep(step.id).execute()
-                            val numberOfSubmissions = submissionsResponse.body().submissions.size
-                            if (submissionsResponse.isSuccessful
-                                    && submissionsResponse.body().submissions.isNotEmpty()
-                                    && submissionsResponse.body().submissions.firstOrNull()?.attempt != attemptId) {
-                                submissionsResponse = api.getSubmissions(attemptId).execute() // if we have another attempt id on server
+                            var submissions = api.getSubmissionForStep(step.id).execute().body()?.submissions
+                            if (submissions?.isNotEmpty() == true && submissions.firstOrNull()?.attempt != attemptId) {
+                                submissions = api.getSubmissions(attemptId).execute()?.body()?.submissions // if we have another attempt id on server
                             }
 
 
-                            if (submissionsResponse.isSuccessful) {
-                                val submission = submissionsResponse.body().submissions.firstOrNull()
+                            if (submissions != null) {
+                                val numberOfSubmissions = submissions.size
+                                val submission = submissions.firstOrNull()
                                 // if null ->  we do not have submissions for THIS ATTEMPT
 
                                 if (submission?.status === Submission.Status.EVALUATION) {
```

Comment:
Is there any sense in `===`?

Reply:
No, it have remained after converting code from java to kotlin. I will change it to just "==".